### ISSUE TITLE
ModuleNotFoundError: No module named 'msgraph'

### ISSUE THREAD

**Issue by laughk** [domainaware/parsedmarc#464]
## Description

I have installed parsedmarc but it does not work with the following error:

```console
$ parsedmarc --help
Traceback (most recent call last):
  File "/tmp/parsedmarc/bin/parsedmarc", line 5, in <module>
    from parsedmarc.cli import _main
  File "/tmp/parsedmarc/lib/python3.11/site-packages/parsedmarc/__init__.py", line 31, in <module>
    from parsedmarc.mail import MailboxConnection
  File "/tmp/parsedmarc/lib/python3.11/site-packages/parsedmarc/mail/__init__.py", line 2, in <module>
    from parsedmarc.mail.graph import MSGraphConnection
  File "/tmp/parsedmarc/lib/python3.11/site-packages/parsedmarc/mail/graph.py", line 10, in <module>
    from msgraph.core import GraphClient
ModuleNotFoundError: No module named 'msgraph'
```

I am seeing this error in at least the following environments

- ArchLinux ( `Linux 5.15.133.1-microsoft-standard-WSL2 #1 SMP Thu Oct 5 21:02:42 UTC 2023 x86_64 GNU/Linux` ), Python 3.11
- Ubuntu 22.04.3 （`Linux 6.2.0-1017-aws #17~22.04.1-Ubuntu SMP Fri Nov 17 21:19:35 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux` ）, Python 3.10

## Steps to reproduce

This occurs when you install parsedmarc and run the command in the following flow.

```console
$ python -m venv venv
$ source venv/bin/activate
(venv) $ python -m pip install parsedmarc
```

Running `parsedmarc` in this state will result in the following

```console
$ parsedmarc
Traceback (most recent call last):
  File "/tmp/parsedmarc/bin/parsedmarc", line 5, in <module>
    from parsedmarc.cli import _main
  File "/tmp/parsedmarc/lib/python3.11/site-packages/parsedmarc/__init__.py", line 31, in <module>
    from parsedmarc.mail import MailboxConnection
  File "/tmp/parsedmarc/lib/python3.11/site-packages/parsedmarc/mail/__init__.py", line 2, in <module>
    from parsedmarc.mail.graph import MSGraphConnection
  File "/tmp/parsedmarc/lib/python3.11/site-packages/parsedmarc/mail/graph.py", line 10, in <module>
    from msgraph.core import GraphClient
ModuleNotFoundError: No module named 'msgraph'

```

## What else I have investigated

I looked up the module name msgraph and found that it refers to the dependent package msgraph-core,
The latest version of msgraph-core, 1.0.0, was released on 2024-01-23, the first major release since 2021-07-27.

- [msgraph-core · PyPI](https://pypi.org/project/msgraph-core/#history)
- [Release v1.0.0 · microsoftgraph/msgraph-sdk-python-core](https://github.com/microsoftgraph/msgraph-sdk-python-core/releases/tag/v1.0.0)

On the other hand, if you look at [project.toml](https://github.com/domainaware/parsedmarc/blob/7d2b431e5f20bdcdb330c4fbb23ce7df5fb0642f/pyproject.toml), you will see that `>= 0.2.2`.

https://github.com/domainaware/parsedmarc/blob/7d2b431e5f20bdcdb330c4fbb23ce7df5fb0642f/pyproject.toml#L50

Therefore, it is assumed that a new installation of parsedmarc installs msgraph-core 1.0.0, and that it no longer works.

## Workaround

It seems to be possible to avoid this problem by specifying a msgraph-core version less than 1.0.0 as follows.

```console
$ python -m venv venv
$ source venv/bin/activate
(venv) $ python -m pip install parsedmarc msgraph-core==0.2.2
```

```console
(venv) $ parsedmarc --help
usage: parsedmarc [-h] [-c CONFIG_FILE] [--strip-attachment-payloads] [-o OUTPUT] [--aggregate-json-filename AGGREGATE_JSON_FILENAME] [--forensic-json-filename FORENSIC_JSON_FILENAME]
                  [--aggregate-csv-filename AGGREGATE_CSV_FILENAME] [--forensic-csv-filename FORENSIC_CSV_FILENAME] [-n NAMESERVERS [NAMESERVERS ...]] [-t DNS_TIMEOUT] [--offline]
                  [-s] [-w] [--verbose] [--debug] [--log-file LOG_FILE] [-v]
                  [file_path ...]

Parses DMARC reports

positional arguments:
  file_path             one or more paths to aggregate or forensic report files, emails, or mbox files'

options:
  -h, --help            show this help message and exit
  -c CONFIG_FILE, --config-file CONFIG_FILE
                        a path to a configuration file (--silent implied)
  --strip-attachment-payloads
                        remove attachment payloads from forensic report output
  -o OUTPUT, --output OUTPUT
                        write output files to the given directory
  --aggregate-json-filename AGGREGATE_JSON_FILENAME
                        filename for the aggregate JSON output file
  --forensic-json-filename FORENSIC_JSON_FILENAME
                        filename for the forensic JSON output file
  --aggregate-csv-filename AGGREGATE_CSV_FILENAME
                        filename for the aggregate CSV output file
  --forensic-csv-filename FORENSIC_CSV_FILENAME
                        filename for the forensic CSV output file
  -n NAMESERVERS [NAMESERVERS ...], --nameservers NAMESERVERS [NAMESERVERS ...]
                        nameservers to query
  -t DNS_TIMEOUT, --dns_timeout DNS_TIMEOUT
                        number of seconds to wait for an answer from DNS (default: 2.0)
  --offline             do not make online queries for geolocation or DNS
  -s, --silent          only print errors
  -w, --warnings        print warnings in addition to errors
  --verbose             more verbose output
  --debug               print debugging information
  --log-file LOG_FILE   output logging to a file
  -v, --version         show program's version number and exit
```

**Comment by tjillez** [comment]
I did all this @jose-m-ferreira but i'm still getting:

parsedmarc.service: Failed with result 'exit-code'.
parsedmarc.service: Main process exited, code=exited, status=1/FAILURE
ModuleNotFoundError: No module named 'msgraph'
from msgraph.core import GraphClient
File "/opt/parsedmarc/venv/lib/python3.10/site-packages/parsedmarc/mail/graph.py", line 10, in <module>
from parsedmarc.mail.graph import MSGraphConnection
JaFile "/opt/parsedmarc/venv/lib/python3.10/site-packages/parsedmarc/mail/__init__.py", line 2, in <module>
from parsedmarc.mail import MailboxConnection
File "/opt/parsedmarc/venv/lib/python3.10/site-packages/parsedmarc/__init__.py", line 31, in <module>
Jafrom parsedmarc.cli import _main
File "/opt/parsedmarc/venv/bin/parsedmarc", line 5, in <module>


pip3 list | grep msgraph
msgraph-core             0.2.2

**Comment by tjillez** [comment]
@jose-m-ferreira  
(venv) root@philip:/home/philip# parsedmarc --version
8.6.4

**Comment by jose-m-ferreira** [comment]
you have to configure a parsedmarc.ini somewhere. i actually haven't been able to get imap out of gmail to work.

**Comment by laughk** [comment]
This issue has been resolved in the following commit (version 8.7.0).

https://github.com/domainaware/parsedmarc/commit/77132b3fc57358da6ea01fa9bb54034254c99c5e
